Add description of https headers for probes

digitalronin opened the pull request:
This came up on [slack](https://mojdt.slack.com/archives/C57UPMZLY/p1551866194097100)

For rails applications with force_ssl = true, probes need extra
http headers. Adding this explanation here will help developers
avoid problems with failing probes, without needing our help.

Commit:
Add description of https headers for probes

For rails applications with force_ssl = true, probes need extra
http headers. Adding this explanation here will help developers
avoid problems with failing probes, without our help.

jabley commented on 99-other-topics/008-zero-downtime-deployment.md:
```diff
@@ -29,5 +29,27 @@ The ability to constantly update your application, with zero downtime, brings a
+
+SSL will be terminated outside of your pod, so your probes will need to respond to HTTP requests. However, Ruby on Rails applications are sometimes configured to only respond to HTTPS traffic (by adding `config.force_ssl = true` in the `config/environments/production.rb` file).
+
+In this case, the application will respond to any HTTP request with a 301 redirect, asking the requester to resend the request to the equivalent HTTPS URL. This will cause your probes to fail, because the redirect will not be followed.
```
Status code varies by Rails version. In 5.0+ it does the Right Thing™ and sometimes sends a 307. Maybe omit?

digitalronin replied:
Good point. I'll update.

Commit:
Remove the exact redirect status code (301)

Rails 5 sends a 307, not a 301. We don't really care which
redirect we're avoiding.
